core/validation/specific_validation/pathfinders/round_1.py

```diff
@@ -186,3 +188,38 @@ def _check_credible_plan_fields(extracted_table_dfs: dict[str, pd.DataFrame]) ->
                         )
                     )
     return error_messages
+
+
+def _check_actual_forecast_reporting_period(extracted_table_dfs: dict[str, pd.DataFrame]) -> list[Message]:
+    reporting_period = extracted_table_dfs["Reporting period"].iloc[0, 0]
+    submission_reporting_period_start_date = PF_REPORTING_PERIOD_TO_DATES_2[reporting_period]["start"]
```

Comment:
do we not already know the submission start and end dates via this bit of code:

https://github.com/communitiesuk/funding-service-design-post-award-data-store/blob/9065c5107bc273ed8b5b86ccf8b0e6b81eeb5c41/core/transformation/pathfinders/round_1/transform.py#L68

Reply:
ie, the dates should be directly accessible from `extracted_table_dfs["Reporting Period"]` in the line above?

Reply:
The bit of code you're referencing is part of transformation which comes after cross-table validation - `extracted_table_dfs` is just a dictionary containing tables extracted directly from the spreadsheet and minimally processed. The flow is:

- Initial validation (high-level checks on auth etc.)
- Table extraction (see `extract_process_validate_tables` in `ingest.py`)
- Table processing (see `extract_process_validate_tables` in `ingest.py`)
- Table validation (see `extract_process_validate_tables` in `ingest.py`)
- Cross-table validation (this file being commented on)
- Transformation
- Database load